### PR TITLE
Add overhead time to junit test case report

### LINKED ISSUE #557: Time info incomplete when using orchestrator
Shard time is calculated by summing all test durations together. FTL doesn't report orchestrator setup/teardown. That means individual test times and overall shard times used by flank are incorrect.

- Work with FTL to get valid time data from the API
- Update sharding algorithm to use real time values

Example JUnit XML result with 3s per test overhead:

```xml
<testsuite name="" tests="2" failures="0" errors="0" skipped="0" time="8s" timestamp="2019-05-13T16:03:42" hostname="localhost">
    <testcase name="test1" classname="com.example.ExampleTest" time="1.0"/>
    <testcase name="test2" classname="com.example.ExampleTest" time="1.0"/>
</testsuite>
```

Total time in the JUnit XML is valid `<testsuite time="8s"`. Individual test times don't include overhead.

### PR DESCRIPTION
**Pull request by jan-gogo**
Fixes #557 

## Checklist

- [x] Unit tested
- [x] release_notes.md updated

### CODE REVIEW

**Comment by bootstraponline** [comment]
awesome!

**Comment by jan-gogo** [comment]
Hey @bootstraponline there is one think to consider here. 
Our new junit report is uploaded to bucket only when option `smart-flank-gcs-path` is used. That means to use smart sharding correctly we need to run flank twice with `smart-flank-gcs-path`. First run will generate and upload correct report so second run can use it to properly calculate shard times. I it correct approach? Or maybe new junit report should be uploaded always, no matter if `smart-flank-gcs-path` used or not.

**Comment by jan-gogo** [comment]
Second thing is that test suites names in junit report may not be unique so method `mergeTestTimes` [here](https://github.com/Flank/flank/blob/f114c83789d95525b35d7daa6f200d7feebd23d6/test_runner/src/main/kotlin/ftl/reports/util/ReportManager.kt#L179) may not work correctly. For now I have not idea how to specify unique names to identify same suites between two test runs. But I guess that we can adjust implementation to omit suite names matching and base only on testcase names (and class names) from all suites. But them also may not be unique if user will add additional test apks where both contains same test name.  Do you have any suggestions?

File: test_runner/src/main/kotlin/ftl/reports/util/ReportManager.kt [hunk 1]
```diff
@@ -176,7 +176,9 @@ object ReportManager {
 
         val oldTestResult = GcStorage.downloadJunitXml(args)
 
-        newTestResult.mergeTestTimes(oldTestResult)
+        if (args.useLegacyJUnitResult) {
+            newTestResult.mergeTestTimes(oldTestResult)
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/reports/util/ReportManager.kt]: I've decided to omit this call cause new junit result (from api) does not favor success over failure nor error.

File: test_runner/src/main/kotlin/ftl/reports/api/CreateTestSuitOverviewData.kt [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
+private fun getOverheadTime(
+    overview: TestSuiteOverview,
+    testCases: List<TestCase>
+) = if (testCases.isEmpty())
+    0.0 else
+    (overview.elapsedTime.millis() - testCases.sumTime()) / testCases.size
+
+private fun List<TestCase>.sumTime(): Double = this
+    .map { it.elapsedTime.millis() }
+    .reduce { acc, d -> acc + d }
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/reports/api/CreateTestSuitOverviewData.kt]: Hey! I think we can simplify this with `this.sumByDouble { it.elapsedTime.millis() }`. Let me know what do you think. Thanks!

**Pull request merged**